### PR TITLE
setup.py: detect "--prefix" correctly.

### PR DESCRIPTION
**Pull request by daym**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 if 'install' in sys.argv:
 	# naively guess where the autostart .desktop file should be installed
-	if '--prefix' in sys.argv or '--home' in sys.argv:
+	if len([x.startswith('--prefix=') for x in sys.argv]) > 0 or '--home' in sys.argv:
```
Comment (Lekensteyn) [inline setup.py]: This should probably just use `any(x.startswith('--prefix') for x in sys.argv)`

Comment (daym) [reply setup.py]: Sure. Added a commit that does that.

File: setup.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
 
 if 'install' in sys.argv:
 	# naively guess where the autostart .desktop file should be installed
-	if '--prefix' in sys.argv or '--home' in sys.argv:
+	if any(x.startswith('--prefix=') for x in sys.argv) or '--home' in sys.argv:
```
Comment (Lekensteyn) [inline setup.py]: You probably want to keep the `--prefix` check (after this change `setup.py --prefix /usr` will not be recognized anymore)

**Comment by jwkokc** [comment]
Wouldn't simply removing the equals sign from line 16 achieve the desired result?

File: setup.py (hunk 2 above)
Comment (daym) [reply setup.py]: For backward compatibility? Good idea.
(Otherwise it's a really strange way to write a long option)

**Comment by daym** [comment]
@jwkokc: It would add false positives. (example: "--prefixdsdewrew")

**Comment by daym** [comment]
@Lekensteyn: I tried to package Solaar for the GNU System Distribution (GuixSD). Its python-build-system uses "--prefix="... for all Python packages. (It's using a directory whose name is a hash value of the source and dependencies as an installation prefix)

The generic installation procedure looks like this:

```
(define* (install #:key outputs (configure-flags '()) use-setuptools?
                  #:allow-other-keys)
  "Install a given Python package."
  (let* ((out (assoc-ref outputs "out"))
         (params (append (list (string-append "--prefix=" out))
                         (if use-setuptools?
                             ;; distutils does not accept these flags
                             (list "--single-version-externally-managed"
                                    "--root=/")
                             '())
                         configure-flags)))
    (call-setuppy "install" params use-setuptools?)
    #t))
```

It's no problem to override it in GuixSD, but we are usually trying to give the patches to upstream in order not to have a big hairy nest of special cases if it can be avoided.

**Comment by Lekensteyn** [comment]
Thanks for the clarification! The reason for asking was to better understand whether this patch is sufficient (or whether it would need more fixes to address future issues).

> It's no problem to override it in GuixSD, but we are usually trying to give the patches to upstream in order not to have a big hairy nest of special cases if it can be avoided.

That's a good idea and improves the ecosystem :-)

**Pull request merged**